src/core/thread_pool.cc

```diff
@@ -54,7 +55,9 @@ void thread_pool::work(sstring name) {
 
             // Prevent the following load of _main_thread_idle to be hoisted before the writes to _completed above.
             std::atomic_thread_fence(std::memory_order_seq_cst);
-            if (_main_thread_idle.load(std::memory_order_relaxed) >= 0) {
+            int64_t idle = _main_thread_idle.load(std::memory_order_relaxed);
+            if (idle >= 0 && idle != last_idle) {
+                last_idle = idle;
```

Comment:
I don't think it's worth the effort. What are the odds that the syscall completes so quickly the reactor isn't woken up? Typically syscalls require I/O and so sleep.  Moreover, typically the reactor isn't idle (if it is, we can affort an extra syscall).

In any case syscalls are moving to io_uring.

Reply:
okay, makes sense.
I'll retract this PR then

Reply:
I forgot process() invokes the system call via the work_item func.
What I had in mind is that this loop just processes results and therefore does very little.